src/sqlfluff/rules/L035.py

```diff
@@ -46,26 +46,24 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
         5.b. We reach the end of case when without matching "NULL": the rule passes
         """
         if context.segment.is_type("case_expression"):
-            fixes: List[LintFix] = []
             for idx, seg in enumerate(context.segment.segments):
-                # When we find ELSE we delete
```

Comment:
The new code is a _little_ shorter, but more significantly, notice we replaced 3 `if` statements with a single `if` statement. 👍